diff -r cc1e666384ea lib_pypy/_ssl_build.py
--- a/lib_pypy/_ssl_build.py	Fri Jan 06 00:01:13 2023 +0200
+++ b/lib_pypy/_ssl_build.py	Sun Jan 08 14:58:05 2023 +0200
@@ -60,27 +60,3 @@
 
 if __name__ == '__main__':
     ffi.compile(verbose=True)
-    if sys.platform == 'win32':
-        # copy dlls from externals to the pwd
-        # maybe we should link to libraries instead of the dlls
-        # to avoid this mess
-        import os, glob, shutil
-        path_parts = os.environ['PATH'].split(';')
-        candidates = [x for x in path_parts if 'externals' in x]
-
-        def copy_from_path(dll):
-            for c in candidates:
-                files = glob.glob(os.path.join(c, dll + '*.dll'))
-                if files:
-                    for fname in files:
-                        print('copying', fname)
-                        shutil.copy(fname, '.')
-                    break
-                else:
-                    print("not copying %s from %s", (dll, c))
-
-        if candidates:
-            for lib in libraries:
-                copy_from_path(lib)
-        else:
-            print('no "externals" on PATH, not copying %s, expect trouble', libraries)
