diff --git a/pypy/tool/release/package.py b/pypy/tool/release/package.py
--- a/pypy/tool/release/package.py
+++ b/pypy/tool/release/package.py
@@ -251,6 +251,9 @@
         try:
             os.symlink(POSIX_EXE, 'pypy')
             os.symlink(POSIX_EXE, 'pypy{}'.format(python_ver))
+            os.symlink(POSIX_EXE, 'python')
+            os.symlink(POSIX_EXE, 'python{}'.format(python_ver))
+            os.symlink(POSIX_EXE, 'python{}'.format(python_ver[1]))
         finally:
             os.chdir(old_dir)
     fix_permissions(pypydir)

