diff -r cc1e666384ea lib_pypy/_sysconfigdata.py
--- a/lib_pypy/_sysconfigdata.py	Fri Jan 06 00:01:13 2023 +0200
+++ b/lib_pypy/_sysconfigdata.py	Sun Jan 08 13:05:57 2023 +0200
@@ -52,7 +52,7 @@
 else:
     build_time_vars['LDLIBRARY'] = 'libpypy3.9-c.so'
     build_time_vars['INCLUDEPY'] = os.path.join(mybase, 'include', 'pypy' + pydot)
-    build_time_vars['LIBDIR'] = os.path.join(mybase, 'bin')
+    build_time_vars['LIBDIR'] = os.path.join(mybase, 'lib')
     build_time_vars['CONFINCLUDEPY'] = build_time_vars['INCLUDEPY']
     # try paths relative to sys.base_prefix first
     tzpaths = [
