--- lib-python/3/distutils/sysconfig_pypy.py	2021-10-21 09:11:53.792196332 -0700
+++ lib-python/3/distutils/sysconfig_pypy.py	2021-10-21 09:12:17.804443956 -0700
@@ -190,10 +190,9 @@
 
         if 'CC' in os.environ:
             newcc = os.environ['CC']
-            if (sys.platform == 'darwin'
-                    and 'LDSHARED' not in os.environ
+            if ('LDSHARED' not in os.environ
                     and ldshared.startswith(cc)):
-                # On OS X, if CC is overridden, use that as the default
+                # If CC is overridden, use that as the default
                 #       command for LDSHARED as well
                 ldshared = newcc + ldshared[len(cc):]
             cc = newcc

