diff -r 1dbd2e1e1866 pypy/tool/release/package.py
--- a/pypy/tool/release/package.py
+++ b/pypy/tool/release/package.py
@@ -478,13 +478,13 @@
     parser.add_argument('--embedded-dependencies', '--no-embedded-dependencies',
                         dest='embed_dependencies',
                         action=NegateAction,
-                        default=(ARCH in ('darwin', 'aarch64', 'x86_64')),
+                        default=False,
                         help='whether to embed dependencies in CFFI modules '
                         '(default on OS X)')
     parser.add_argument('--make-portable', '--no-make-portable',
                         dest='make_portable',
                         action=NegateAction,
-                        default=(ARCH in ('darwin',)),
+                        default=False,
                         help='make the package portable by shipping '
                             'dependent shared objects and mangling RPATH')
     options = parser.parse_args(args)

