From ac21be539816ae0bd43cd7fc41ea29f6ec1be826 Mon Sep 17 00:00:00 2001
Date: Sun, 4 Feb 2024 23:11:15 +0200
Subject: [PATCH] remove top level __init__.py (issue 4885)

---
 pypy/tool/release/package.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pypy/tool/release/package.py b/pypy/tool/release/package.py
index 95159d03c0f..5c626528d42 100755
--- a/pypy/tool/release/package.py
+++ b/pypy/tool/release/package.py
@@ -358,6 +358,8 @@ def create_package(basedir, options, _fake=False):
                                            '*.lib', '*.exp', '*.manifest', '__pycache__'))
     for file in ['README.rst',]:
         shutil.copy(str(basedir.join(file)), str(pypydir))
+    for file in ['__init__.py',]:
+        os.unlink(str(target / file))
     # Use original LICENCE file
     base_file = str(basedir.join('LICENSE'))
     with open(base_file) as fid:
-- 
2.34.1

