--- a/rpython/rlib/clibffi.py	Fri Oct 15 14:59:32 2021 +0200
+++ b/rpython/rlib/clibffi.py	Mon Oct 18 07:36:46 2021 +0300
@@ -107,7 +107,7 @@
     # libffidir = py.path.local(cdir).join('src', 'libffi_msvc')
     eci = ExternalCompilationInfo(
         includes = ['ffi.h', 'windows.h'],
-        libraries = ['kernel32', 'libffi-7'],
+        libraries = ['kernel32', 'libffi'],
         separate_module_sources = separate_module_sources,
         post_include_bits = post_include_bits,
         )
--- a/rpython/translator/driver.py	Fri Oct 15 14:59:32 2021 +0200
+++ b/rpython/translator/driver.py	Mon Oct 18 07:36:46 2021 +0300
@@ -507,11 +507,6 @@
                         newname = newexename.new(basename=soname.basename)
                         shutil.copyfile(str(name), str(newname.new(ext=ext)))
                         self.log.info("copied: %s" % (newname,))
-                    # HACK: copy libcffi-7.dll which is required for venvs
-                    # At some point, we should stop doing this, and instead
-                    # use the artifact from packaging the build instead
-                    libffi = py.path.local.sysfind('libffi-7.dll')
-                    shutil.copyfile(str(libffi), os.getcwd() + r'\libffi-7.dll')
             self.c_entryp = newexename
         self.log.info("created: %s" % (self.c_entryp,))

