Author: Isuru Fernando <isuruf@gmail.com>

--- ./pypy3/lib_pypy/_sysconfigdata.py      2021-10-21 09:53:52.246898790 -0700
+++ ./pypy3/lib_pypy/_sysconfigdata.py  2021-10-21 09:54:03.499017441 -0700
@@ -6,7 +6,7 @@
 
 so_ext = _imp.extension_suffixes()[0]
 
-mybase = os.path.dirname(os.path.dirname(__file__))
+mybase = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
 
 # FIXME: The compiler-specific values should be exported from the build,
 # perhaps via a special __pypy__._sysconfigdata module?

