From b07042c95b76f9c4f67369d4d9ca9cda4546f647 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Wed, 27 Oct 2021 15:22:13 +1100
Subject: [PATCH vopenssl3] also patch _ssl_build.py

---
 lib_pypy/_ssl_build.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib_pypy/_ssl_build.py b/lib_pypy/_ssl_build.py
index ed59270..74f9eb1 100644
--- a/lib_pypy/_ssl_build.py
+++ b/lib_pypy/_ssl_build.py
@@ -17,7 +17,8 @@ ffi = build_ffi_for_binding(
     modules=[
         # This goes first so we can define some cryptography-wide symbols.
         "cryptography",
-
+        # Provider comes early as well so we define OSSL_LIB_CTX
+        "provider",
         "aes",
         "asn1",
         "bignum",
@@ -25,7 +26,6 @@ ffi = build_ffi_for_binding(
         "cmac",
         "conf",
         "crypto",
-        "ct",
         "dh",
         "dsa",
         "ec",
-- 
2.32.0.windows.2

